crates/uv-interpreter/src/discovery.rs

```diff
@@ -44,9 +44,15 @@ pub enum InterpreterRequest {
 /// The sources to consider when finding a Python interpreter.
 #[derive(Debug, Clone, PartialEq, Eq, Default)]
 pub enum SourceSelector {
+    // Consider all interpreter sources.
     #[default]
     All,
-    Some(HashSet<InterpreterSource>),
+    // Only consider system interpreter sources
+    System,
+    // Only consider virtual environment sources
+    VirtualEnv,
```

Comment:
So this is effectively shorthand for a specific set of sources, and lets us add a better error message later?

Reply:
Yeah. I actually don't think there's a use-case for a specific set of sources outside of testing. Instead of enumerating all the possible sources, we'll list the sources that we considered in a way that is actually relevant to the user.

I think there are some other directions we could go in the future, like ruling out some sources given the request kind but this was a simple path forward for this release.